OpenRA.Mods.Common/Traits/Demolition.cs

```diff
@@ -98,20 +101,39 @@ public string VoicePhraseForOrder(Actor self, Order order)
 
 		class DemolitionOrderTargeter : UnitOrderTargeter
 		{
-			public DemolitionOrderTargeter(string cursor)
-				: base("C4", 6, cursor, true, false) { }
+			readonly DemolitionInfo info;
+
+			public DemolitionOrderTargeter(DemolitionInfo info)
+				: base("C4", 6, info.Cursor, true, false)
```

Comment:
IMO the best solution here would be to pass the `TargetStances` and `ForceTargetStances` into `UnitOrderTargeter`, and then keep a legacy constructor that takes `targetEnemyUnits` and `targetAllyUnits` and converts them to equivalent stance enums so that you don't need to rewrite all the other subclasses here.